app/shop.py

```diff
@@ -0,0 +1,23 @@
+class Shop:
+    def __init__(self, name: str, location: list, products: dict) -> None:
+        self.name = name
+        self.location = location
+        self.products = products
+
+    def print_receipt(self, customer: dict) -> None:

```

Comment:
The `customer` parameter should be an instance of the `Customer` class, not a dictionary. This allows you to access attributes like `name` and `product_cart` directly from the `Customer` instance.